### PR TITLE
Todo page(use prams)

### PR DESCRIPTION
**Pull request by Amin-zare**

### CODE REVIEW

File: src/components/todos/TodoItem.tsx [hunk 1]
```diff
@@ -17,6 +18,7 @@ const TodoItem: React.FC<Props> = ({
 }) => {
   const [isEditing, setIsEditing] = useState(false)
   const [newTitle, setNewTitle] = useState(todo.title)
+  const editInputRef = React.createRef<HTMLInputElement>()
```
Comment (behnum) [inline src/components/todos/TodoItem.tsx:21]: اینطوری editInputRef در واقع بهتر پیاده می‌شه:
۱. از روی input برش داشتیم.
۲. اینجا از هوک useRef استفاده کردیم.
۳. لینک کردیمش به input.

File: src/routes/todo/TodoPage.tsx [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        </h2>
+        <div className='card-body'>
+          <p className='card-text'>
+            Todo item with ID:{' '}
+            <span className='font-weight-bold'>
+              {params.id || 'Not Provided'}
+            </span>
+          </p>
+          <p className='card-text'>
+            Status:
```
Comment (behnum) [inline src/routes/todo/TodoPage.tsx:85]: بعنوان پیشنهاد و راه حل:
&nbsp; اینجا لازم هست.

```suggestion
            Status:&nbsp;
```
بهش میگن:
Non-breaking space - Wikipedia
https://en.wikipedia.org/wiki/Non-breaking_space

File: src/routes/todo/TodoPage.tsx [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+                    const node = titleRef.current
+                    if (node) {
+                      const selection = window.getSelection() // Get the current selection object. This represents what's currently selected/highlighted in the browser.
+                      const range = document.createRange() // Create a new "range". A range represents a portion of a document.
+                      range.selectNodeContents(node) // Set the range to select the entire content of our h2 element
+                      range.collapse(false) // false means collapse to the end
+                      selection?.removeAllRanges() // Clear any existing selections.
+                      selection?.addRange(range) // Apply our new range, which effectively moves the cursor to the end of the h2 content
+                    }
+                  }, 0)
```
Comment (behnum) [inline src/routes/todo/TodoPage.tsx:138]: Quick question! If delay is 0 ms, let's discuss the purpose of this setTimeout here. (#later)

**Review by behnum — COMMENTED** [review]
Interesting stuff. Let's discuss them further over the coming days.

File: src/routes/todo/TodoPage.tsx (hunk 3 above)
Comment (Amin-zare) [reply src/routes/todo/TodoPage.tsx:138]: The use of setTimeout with a delay of 0ms might seem unintuitive, but it has a specific purpose in JavaScript: it effectively defers the execution of the provided function until the current call stack is clear. This technique is often used to ensure that some code runs after the current execution context completes and after the browser has had an opportunity to update the DOM or UI, even if it's just milliseconds later.

Comment (behnum) [reply src/routes/todo/TodoPage.tsx:138]: > Good one 👍🏻
javascript - Why is setTimeout(fn, 0) sometimes useful? - Stack Overflow
https://stackoverflow.com/questions/779379/why-is-settimeoutfn-0-sometimes-useful

**Pull request merged**